.github/workflows/build-push.yml

```
@@ -0,0 +1,63 @@
+        with:
+          path: /var/lib/docker
+          key: ${{ runner.os }}-docker-${{ hashFiles('**/windows.Dockerfile' && '**/linux.Dockerfile') }}
+          
+      - name: Build and Push Docker Images
+        run: npm run build -- ${{ inputs.no_cache && '--no-cache' || '' }}
+
+      - name: Login to Docker Hub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v3
+        with:
+            username: ${{ secrets.DOCKERHUB_USERNAME }}
```

YAMLlint (1.35.1)

[warning] wrong indentation: expected 10 but found 12

(indentation)